File: src/main/java/com/example/eyeofsauron/service/CheckPointService.java
package com.example.eyeofsauron.service;

import com.example.eyeofsauron.entity.CheckPoint;
import com.example.eyeofsauron.repository.CheckPointRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

/**
 * Сервис для работы с контрольными точками
 *
 * @author rodkinsi
 */
@Service
public class CheckPointService {
    private CheckPointRepository checkPointRepository;

    public List<CheckPoint> getAll() { return checkPointRepository.findAll(); }

    public Optional<CheckPoint> getById(Long id) { return checkPointRepository.findById(id); }

    public void create(CheckPoint checkPoint) { checkPointRepository.save(checkPoint); }

    public void update(CheckPoint checkPoint) { checkPointRepository.save(checkPoint); }

    public void delete(CheckPoint checkPoint) { checkPointRepository.delete(checkPoint); }

    public void deleteById(Long id) { checkPointRepository.deleteById(id); }
}

File: src/main/kotlin/com/example/eyeofsauron/service/CheckPointService.kt
package com.example.eyeofsauron.service

import com.example.eyeofsauron.entity.CheckPoint
import com.example.eyeofsauron.repository.CheckPointRepository
import org.springframework.stereotype.Service

/**
 * Сервис для работы с контрольными точками
 * @author rodkinsi
 */
@Service
class CheckPointService(private val repository: CheckPointRepository) {
    fun getAll(): List<CheckPoint> = repository.findAll()

    fun getById(id: Long) = repository.findById(id)

    fun getByBypass(bypassIds: List<Long>) = repository.findByRouteBypass_IdIn(bypassIds)

    fun create(checkPoint: CheckPoint) = repository.save(checkPoint)

    fun update(checkPoint: CheckPoint) {
        if(!repository.existsById(checkPoint.id))
            throw(Exception("Record not found!"))
        repository.save(checkPoint)
    }

    fun delete(checkPoint: CheckPoint) = repository.delete(checkPoint)

    fun deleteById(id: Long) = repository.deleteById(id)
}